batch/batch/server/server.py

```diff
@@ -113,51 +113,50 @@ def _next_task(self):
         self._task_idx += 1
         if self._task_idx < len(self._tasks):
             self._current_task = self._tasks[self._task_idx]
-        else:
-            self._delete_pvc()
 
     def _has_next_task(self):
         return self._task_idx < len(self._tasks)
 
-    def pvc(self):
-        if self._pvc is None:
-            self._pvc = v1.create_namespaced_persistent_volume_claim(
-                POD_NAMESPACE,
-                kube.client.V1PersistentVolumeClaim(
-                    metadata=kube.client.V1ObjectMeta(
-                        generate_name=f'job-{self.id}-',
-                        labels={'app': 'batch-job',
-                                'hail.is/batch-instance': instance_id}),
-                    spec=kube.client.V1PersistentVolumeClaimSpec(
-                        access_modes=['ReadWriteOnce'],
-                        volume_mode='Filesystem',
-                        resources=kube.client.V1ResourceRequirements(
-                            requests={'storage': POD_VOLUME_SIZE}),
-                        storage_class_name=STORAGE_CLASS_NAME)),
-                _request_timeout=KUBERNETES_TIMEOUT_IN_SECONDS)
-            log.info(f'created pvc name: {self._pvc.metadata.name} for job {self.id}')
-        return self._pvc
+    def _create_pvc(self):
+        pvc = v1.create_namespaced_persistent_volume_claim(
+            POD_NAMESPACE,
+            kube.client.V1PersistentVolumeClaim(
+                metadata=kube.client.V1ObjectMeta(
+                    generate_name=f'job-{self.id}-',
+                    labels={'app': 'batch-job',
+                            'hail.is/batch-instance': instance_id}),
+                spec=kube.client.V1PersistentVolumeClaimSpec(
+                    access_modes=['ReadWriteOnce'],
+                    volume_mode='Filesystem',
+                    resources=kube.client.V1ResourceRequirements(
+                        requests={'storage': POD_VOLUME_SIZE}),
+                    storage_class_name=STORAGE_CLASS_NAME)),
+            _request_timeout=KUBERNETES_TIMEOUT_IN_SECONDS)
+        log.info(f'created pvc name: {self._pvc.metadata.name} for job {self.id}')
+        return pvc
 
     def _create_pod(self):
-        assert not self._pod_name
+        assert self._pod_name is not None
```

Comment:
Do you mean is None? If the pod exists, you probably don't mean to re-create it?

Reply:
yep